perfkitbenchmarker/providers/gcp/gce_network.py

```diff
@@ -41,6 +44,425 @@
+    if route_name not in self.routes:
+      self.routes[route_name] = GceRoute(route_name, dest_cidr, self.network_name, next_hop_tun, self.region, self.project, self.region)
+      self.routes[route_name].Create()
+
+  def Create(self):
+    """Creates the actual VPNGW."""
+    benchmark_spec = context.GetThreadBenchmarkSpec()
+    if benchmark_spec is None:
+      raise errors.Error('GetNetwork called in a thread without a '
+                         'BenchmarkSpec.')
+    if self.created:
+      return
+    self.vpn_gateway_resource.Create()
+
+    self.created = True
+
+  def Delete(self):
+    """Deletes the actual VPNGW."""
+    if self.ip_resource:
+      self.ip_resource.Delete()
+
+    if self.tunnels:
+      vm_util.RunThreaded(lambda tun: self.tunnels[tun].Delete(),
+                          list(self.tunnels.keys()))
+
+    if self.forwarding_rules:
+      vm_util.RunThreaded(lambda fr: self.forwarding_rules[fr].Delete(),
+                          list(self.forwarding_rules.keys()))
+
+    if self.routes:
+      vm_util.RunThreaded(lambda route: self.routes[route].Delete(),
+                          list(self.routes.keys()))
+
+    if self.vpn_gateway_resource:
+      self.vpn_gateway_resource.Delete()
+
+    self.created = False
+
+
+class GceVPNGWResource(resource.BaseResource):
+  """Object representing a GCE VPN Gateway Resource."""
+
+  def __init__(self, name, network_name, region, cidr, project):
+    super(GceVPNGWResource, self).__init__()
+    self.name = name
+    self.network_name = network_name
+    self.region = region
+    self.cidr = cidr
+    self.project = project
+
+  def _Create(self):
+    cmd = util.GcloudCommand(self, 'compute', 'target-vpn-gateways', 'create',
+                             self.name)
+    cmd.flags['network'] = self.network_name
+    cmd.Issue()
+
+  def _Exists(self):
+    cmd = util.GcloudCommand(self, 'compute', 'target-vpn-gateways', 'describe',
+                             self.name)
+    _, _, retcode = cmd.Issue(suppress_warning=True, raise_on_failure=False)
+    return not retcode
+
+  def _Delete(self):
+    cmd = util.GcloudCommand(self, 'compute', 'target-vpn-gateways', 'delete',
+                             self.name)
+    cmd.Issue(raise_on_failure=False)
+
+
+class GceIPAddress(resource.BaseResource):
+  """Object representing a GCE IP address."""
+
+  def __init__(self, project, region, name):
+    super(GceIPAddress, self).__init__()
+    self.project = project
+    self.region = region
+    self.name = name
+    self.ip_address = None
+
+  def _Create(self):
+    """ Allocates a public IP for the VPN GW """
+    cmd = util.GcloudCommand(self, 'compute', 'addresses', 'create', self.name)
+    cmd.Issue()
+
+  def _PostCreate(self):
+    cmd = util.GcloudCommand(self, 'compute', 'addresses', 'describe', self.name)
+    cmd.flags['format'] = 'value(address)'
+    stdout, _, _ = cmd.Issue()
+    self.ip_address = stdout.encode('ascii', 'ignore').rstrip().decode()
```

Comment:
What kind of output is this giving that it requires the encode/decode?

Reply:
This started returning unicode strings (u'1.2.3.4') a while back so I added the encode. When I did the most recent rebase it started kicking back binary (b'1.2.3.4') so I added the decode.

Reply:
Ok - it's working now with just stdout.rstrip().